impl/src/main/java/org/kuali/rice/krad/service/impl/DocumentServiceImpl.java

```diff
@@ -714,7 +716,7 @@ public Document getByDocumentHeaderIdSessionless(String documentHeaderId) throws
             LOG.debug("Retrieving doc id: " + documentHeaderId + " from workflow service.");
         }
 
-        Person person = getPersonService().getPersonByPrincipalName(KRADConstants.SYSTEM_USER);
+        Person person = getPersonService().getSystemUserPersonFromDb();

```

Comment:
This method call will prevent going to EDS and asking for a non existent system user 'kr', instead will get it from the kim tables.